File: ios-SpaceSub/ios-SpaceSub/Models/BankConnection.swift
import Foundation

nonisolated enum BankProvider: String, Codable, Sendable {
    case flex = "FLEX"
}

nonisolated enum BankConnectionStatus: String, Codable, Sendable {
    case connected = "CONNECTED"
    case expired = "EXPIRED"
    case error = "ERROR"
    case disconnected = "DISCONNECTED"
}

nonisolated struct BankConnection: Codable, Identifiable, Sendable {
    let id: String
    let provider: BankProvider
    let status: BankConnectionStatus
    let lastSyncAt: String?
    let expiresAt: String?
    let createdAt: String
}

nonisolated struct BankOAuthURLResponse: Codable, Sendable {
    let url: String
}

nonisolated struct SyncResult: Codable, Sendable {
    let ok: Bool
    let provider: BankProvider
    let imported: Int
    let accounts: Int
}

nonisolated struct ConnectByCodeRequest: Codable, Sendable {
    let code: String
}

nonisolated struct ConnectByCodeResponse: Codable, Sendable {
    let ok: Bool
    let provider: BankProvider
}

File: android-SpaceSub/app/src/main/java/dev/squad52/spacesub/models/BankConnection.kt
package dev.squad52.spacesub.models

import com.google.gson.annotations.SerializedName

enum class BankConnectionStatus {
    @SerializedName("CONNECTED") CONNECTED,
    @SerializedName("EXPIRED") EXPIRED,
    @SerializedName("ERROR") ERROR,
    @SerializedName("DISCONNECTED") DISCONNECTED
}

data class BankConnection(
    val id: String,
    val provider: String,
    val status: BankConnectionStatus,
    val lastSyncAt: String? = null,
    val expiresAt: String? = null,
    val createdAt: String
)

data class BankOAuthURLResponse(val url: String)

data class SyncResult(
    val ok: Boolean,
    val provider: String,
    val imported: Int,
    val accounts: Int
)

data class ConnectByCodeRequest(val code: String)

data class ConnectByCodeResponse(
    val ok: Boolean,
    val provider: String
)
